tests/aggregation_regress/tests.py

```diff
@@ -1507,6 +1507,12 @@ class DistinctAggregate(Aggregate):
             allow_distinct = True
         DistinctAggregate('foo', distinct=True)
 
+    def test_filter_on_numerical_aggregate(self):
+        try:
+            list(Publisher.objects.annotate(good_length=Avg('book__pages', filter=Q(book__rating__gte=4.0))))
```

Comment:
Just wondering, should this test also check the value of this result, and check both `.annotate()` and `.aggregate()`? I know it's not the specific regression we're fixing here, but if there doesn't already exist a regression test for numerical aggregates with filters, maybe it should go the whole way, similarly to other tests in this file?

Reply:
I would move test to the `tests/aggregation/test_filter_argument.py`, and yes, we should check the result instead of checking that exception is not raised, e.g.:
```python
def test_filtered_numerical_aggregates(self):
    for aggregate, expected_result in (
        (Avg, Approximate(66.7, 1)),
        (StdDev, Approximate(24.9, 1)),
        (Variance, Approximate(622.2, 1)),
   ):
       with self.subTest(aggregate=aggregate.__name__):
           agg = aggregate('age', filter=Q(name__startswith='test'))
           self.assertEqual(Author.objects.aggregate(age=agg)['age'], expected_result)
```

Reply:
Done, thank you @felixxm. @czlee: Don't think it would be necessary to check both `.annotate()` and `.aggregate()` as it is a test for the arithmetic operation, which would have the same behaviour in either case.